src/adm/filter.rs

```diff
@@ -144,7 +146,7 @@ impl AdmFilter {
     }
 
     /// check to see if the bucket has been modified since the last time we updated.
-    pub async fn requires_update(&self) -> HandlerResult<bool> {
+    pub async fn requires_update(&self, req: reqwest::Client) -> HandlerResult<bool> {
```

Comment:
nit: Could avoid the clone in the updated closure
```suggestion
    pub async fn requires_update(&self, req: &reqwest::Client) -> HandlerResult<bool> {
```